File: Java/InterfaceTest1/src/CurrentAccount.java
package banking;

//final class cannot be subclassed
final class CurrentAccount extends Account {

	public void deposit(double amount) {
		if(balance < 0)
			amount *= 0.9; //charging 10% fine
		balance += amount;
	}

	public void withdraw(double amount) throws InsufficientFundsException{
		balance -= amount;
	}
}

File: DotNet/Inheritance/InterfaceTest1/Banking/CurrentAccount.cs
namespace Banking;

//sealed class cannot have derived classes
sealed class CurrentAccount : Account, IFineable
{
    public override void Deposit(decimal amount)
    {
        Balance += amount;
    }

    public override void Withdraw(decimal amount)
    {
        Balance -= amount;
    }

    //explicit interface implementation
    bool IFineable.Withdraw(decimal fine)
    {
        if(Balance < 0)
        {
            Balance -= fine;
            return true;
        }
        return false;
    }
}